### DISCUSSION TITLE
Integrate with go-storage to allow using various storage service as storing location

### DISCUSSION THREAD

**Discussion by Xuanwo** [filecoin-project/lotus#6218] [Ideas]
## Background

[go-storage](https://github.com/aos-dev/go-storage) is an application-oriented unified storage layer for Golang which allows the user to connect various storage services with the same interface. For now, go-storage supports s3, gcs, azblob, oss, cos, and more supports are on the way: https://github.com/aos-dev/go-storage/issues/536.

## Proposal

I propose to integrate lotus with go-storage to allow using various storage service as storing location.

After this proposal implemented, we can add s3 service (gcs, azblob, minio and any other object storage services):

```shell
lotus-miner storage attach --init --store s3://bucket_name/path
```

## Implementations

Need Support

## Future

go-storage intends to build a unified storage layer, so we are not limited to object storage services, [dropbox](https://github.com/aos-dev/go-service-dropbox), [ipfs](https://github.com/aos-dev/go-service-ipfs) is also on our roadmap.

After these services implemented, we can:

```shell
lotus-miner storage attach --init --store ipfs://node_id/path
```

Looks interesting?

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
what is the advantage over a locally mounted folder?
